errors/errors.go

```diff
@@ -0,0 +1,195 @@
+package errors
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// InternalErr represents a general case issue that cannot be
+	// categorized as any of the below cases.
+	InternalErr = NewRootError(0, "internal")
```

Comment:
This approach looks good to me, this is similar to what I had in mind when saying that we might be better off with structs that would map err code to some sort of generic description.

I'd argue that RootError is not a good name though, I'd rather settle for just Error{}.
I would also argue that New() in case of already instantiated error is no the best method name either, so I'd prefer it to be Wrap as well, that would be way more consistent. 

I still think that the migration has to be done in one big PR without introducing extra abstractions for transition period, cause that's simply postponing the inevitable and our codebase is not so huge.